R/data.table.R

```diff
@@ -2022,7 +2022,12 @@ as.list.data.table <- function(x, ...) {
 
 dimnames.data.table <- function(x) {
     if (!cedta()) {
-        if (!identical(class(x),c("data.table","data.frame"))) stop("data.table inherits from data.frame (from v1.5) but this data.table does not. Has it been created manually (e.g. by using 'structure' rather than 'data.table') or saved to disk using a prior version of data.table? The correct class is c('data.table','data.frame').")
+        if (!inherits(x, "data.frame")) 
```

Comment:
@arunsrinivasan I think the fact that it gets here in the first place (unless a user invokes `data.table:::dimnames.data.table` directly) implies that `inherits(x, "data.table")` is `TRUE`, since something dispatched to a `data.table` method. Am I mistaken?

This is why I think we can ditch the test altogether -- seems this worry about `x` being `data.table` but not `data.frame` is pretty obsolete. 

If we _do_ test, we should test for `data.frame` so we're sure `dimnames.data.frame` is proper to dispatch.